### ISSUE TITLE
Missing startval with primitive data type as json

### ISSUE THREAD

**Issue by Gerinjo** [rodikh/angular-json-editor#28]
Hi, 

my schema looks like:

```
{ "title" : "Integer Value",
  "type" : "number"
}
```

The startval should be e.g. 55.

But the generated input field of the json editor always contains 0. I debugged inside the jsonEditor directive definition and I found out, that the startval is not set, if the value is not an object.

```
if (angular.isObject(scope.startval)) {
                // Support both $http (i.e. $q) and $resource promises, and also normal object.
                startValPromise = $q.when(scope.startval);
}
```

I changed it to

```
if (angular.isDefined(scope.startval)) {
                // Support both $http (i.e. $q) and $resource promises, and also normal object.
                startValPromise = $q.when(scope.startval);
            }
```

Now it works fine.

Hope it helps,
Best regards,
Jerome

**Comment by rodikh** [comment]
Is your form just one field?
Will changing the schema to an object with only one property also work?

I'll run a few tests and use your solution if everything checks out.
thanks.

**Comment by rodikh** [comment]
should be solved in 0.3.0
